qiskit/providers/aer/backends/aerbackend.py

```diff
@@ -146,28 +154,68 @@ def run(self,
                 DeprecationWarning,
                 stacklevel=3)
 
-        if isinstance(circuits, (QasmQobj, PulseQobj)):
-            warnings.warn('Using a qobj for run() is deprecated and will be '
-                          'removed in a future release.',
-                          PendingDeprecationWarning,
-                          stacklevel=2)
-            qobj = circuits
+        executor = None
+        if backend_options and "executor" in backend_options:
+            executor = backend_options["executor"]
+            del backend_options["executor"]
+
+        if "executor" in run_options:
+            executor = run_options["executor"]
+            del run_options["executor"]
```

Comment:
Instead of doing this why not just add `executor` to the default options? That will also enable users to something like:

```
sim_backend = AerSimulator(executor=dask_magic())
sim_backend.run([qc]*10**5)
sim.backend.run([new_qc]*10**4)
```

and have it all run on dask.

Reply:
At first, I added the executor option as your code but DASK client variable is needed to be a local variable for the serialization. So I moved the executor option as a run option.

Reply:
```
class AerBackend(Backend, ABC):
    def __init__():
        self.executor = None:
   /* ... */
    def run():
        self.executor = getattr(self._options, 'executor')
```
This code occurs a cloudpickle error because the executor is not a local variable. The same error is mentioned in this [website](https://stackoverflow.com/questions/45976880/pickle-error-when-submitting-task-using-dask)

```
class AerBackend(Backend, ABC):
   /* ... */
    def run():
       executor = None
       executor = getattr(self._options, 'executor')
       delattr(self._options, 'executor')
```
I changed an instance variable to a local variable of `executor` before. This code can be executed when the first call of `run` function but the second call is failed because `delattr` is called. If `delattr` is not called, `Failed to load qobj: to_json not implemented for this type of object: <class 'concurrent.futures.thread.ThreadPoolExecutor'>` occurs. 

So I changed the `executor` option as the run option because `executor` is passed every calling `run` function. Are there any good idea to realize the `executor` options as a config option?

Reply:
I found that the following code can be executed. I don't know why this code is ok but I can move `executor` option to default options. 
```
class AerBackend(Backend, ABC):
    def __init__():
        self.executor = None:
   /* ... */
    def run():
         executor = None
        if self._executor:
           executor = self._executor
        else:
           executor = getattr(self._options, 'executor')
           delattr(self._options, 'executor')
       /* ....*/     
       aer_job_set = AerJobSet(self, job_id, self._run, experiments, executor)
       aer_job_set.submit()
       self._executor = executor
       return aer_job_set
```